packages/plugin-kit/src/source-code.js

```diff
@@ -243,12 +287,30 @@ export class TextSourceCodeBase {
 	 * @param {Object} options The options for the instance.
 	 * @param {string} options.text The source code text.
 	 * @param {Options['RootNode']} options.ast The root AST node.
-	 * @param {RegExp} [options.lineEndingPattern] The pattern to match lineEndings in the source code.
+	 * @param {RegExp} [options.lineEndingPattern] The pattern to match lineEndings in the source code. Defaults to `/\r?\n/gu`.
+	 * @param {0|1} [options.lineStart] The line number at which the parser starts counting. Defaults to `1` for ESTree compatibility.
+	 * @param {0|1} [options.columnStart] The column number at which the parser starts counting. Defaults to `0` for ESTree compatibility.
 	 */
-	constructor({ text, ast, lineEndingPattern = /\r?\n/u }) {
+	constructor({
+		text,
+		ast,
+		lineEndingPattern = /\r?\n/gu,
+		lineStart = 1,
+		columnStart = 0,
+	}) {
 		this.ast = ast;
 		this.text = text;
-		this.#lines = text.split(lineEndingPattern);
+		this.#lineStart = lineStart;
+		this.#columnStart = columnStart;
```

Comment:
I don't think we need to pass these in. We should be able to get this information off of `ast`.